### PR TITLE
Machine-to-machine auth post draft.

### PR DESCRIPTION
**Pull request by sebadoom**
Icon to be determined.

screenshot_2018-07-04 using machine-to-machine m2m authorization

### CODE REVIEW

File: _posts/2018-06-14-using-m2m-authorization.markdown [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+The [client credentials grant] from OAuth 2.0 attempts to fulfill the need for these scenarios. In the client credentials grant, the client holds two pieces of information: the client ID and the client secret. With this information, the client can request an access token for a protected resource.
+
+![Client Credentials Grant](https://cdn.auth0.com/docs/media/articles/api-auth/client-credentials-grant.png)
+
+1. The client makes a request to the authorization server sending the `client ID`, the `client secret`, along with the `audience` and `scopes` claims.
+2. The authorization server validates the request, and, if successful, sends a response with an access token.
```
Comment (aaguiarz) [inline _posts/2018-06-14-using-m2m-authorization.markdown]: Actually, for client credentials the scopes should not be sent, and you always get granted the ones specified in the server when you authorize the M2M app to call an API.

File: _posts/2018-06-14-using-m2m-authorization.markdown [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+The [client credentials grant] from OAuth 2.0 attempts to fulfill the need for these scenarios. In the client credentials grant, the client holds two pieces of information: the client ID and the client secret. With this information, the client can request an access token for a protected resource.
+
+![Client Credentials Grant](https://cdn.auth0.com/docs/media/articles/api-auth/client-credentials-grant.png)
+
+1. The client makes a request to the authorization server sending the `client ID`, the `client secret`, along with the `audience` and `scopes` claims.
```
Comment (aaguiarz) [inline _posts/2018-06-14-using-m2m-authorization.markdown]: We are using 'client' and now in the Dashboard we use 'Application' for the same purpose. Shouldn't use 'application'?

File: _posts/2018-06-14-using-m2m-authorization.markdown [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+```
+POST https://<YOUR_AUTH0_DOMAIN>/oauth/token
+Content-Type: application/json
+{
+  "audience": "<API_IDENTIFIER>",
+  "grant_type": "client_credentials",
+  "client_id": "<YOUR_CLIENT_ID>",
+  "client_secret": "<YOUR_CLIENT_SECRET>",
+  "scope": "write:logs" // Optional
+}
```
Comment (aaguiarz) [inline _posts/2018-06-14-using-m2m-authorization.markdown]: No scope

File: _posts/2018-06-14-using-m2m-authorization.markdown [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+{
+  "access_token": "eyJz93a...k4laUWw",
+  "token_type": "Bearer",
+  "expires_in": 86400
+}
+```
+
+In Auth0, to use the client credentials grant, you must first enable the grant through the dashboard:
```
Comment (aaguiarz) [inline _posts/2018-06-14-using-m2m-authorization.markdown]: We have an M2M application type, shouldn't we use that one for the sake of the article? We can also clarify that any confidential client (webapps and m2m apps) can use the client_credentials grant

Also, I think we should show the new flow here (create the m2m app, select an API, authorize it, etc), the new flow is described here https://auth0team.atlassian.net/wiki/spaces/PROD/blog/2018/05/31/91947046/Improvements+to+Machine+to+Machine+UX.

Another option is to link to docs that are also in a docs PR https://github.com/auth0/docs/pull/6222

File: _posts/2018-06-14-using-m2m-authorization.markdown [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+![Client Credentials Grant in the Auth0 Dashboard](https://cdn.auth0.com/blog/m2m/2-dashboard.png)
+
+This toggle is located in `Applications` -> `<Your App Name>` -> `Settings` -> `Advanced Settings` -> `Grant Types`.
+
+The `audience` claim is related to the API you are requesting access to. You can create audiences for your APIs and protected resources in the [API](https://manage.auth0.com/#/apis) section of the Auth0 Dashboard.
+
+### Scopes and Granular Permissions in Auth0
+Thanks to the use of [rules in Auth0](https://auth0.com/docs/rules/current), it is very easy to have granular permissions for machine-to-machine communications. When a client uses the client credentials grant, a rule can be run to check for any data in the request, including `scopes` or `roles`. With this information, you can choose to either grant or deny the request.
+
```
Comment (aaguiarz) [inline _posts/2018-06-14-using-m2m-authorization.markdown]: We can explain that you can grant specific scopes to the M2M application.

File: _posts/2018-06-14-using-m2m-authorization.markdown (hunk 4 above)
Comment (aaguiarz) [reply _posts/2018-06-14-using-m2m-authorization.markdown]: Also, make sure you have a new tenant, with non legacy-grants enabled, as the grants dialog will be cleaner

File: _posts/2018-06-14-using-m2m-authorization.markdown (hunk 2 above)
Comment (sebadoom) [reply _posts/2018-06-14-using-m2m-authorization.markdown]: In this case I consider the use of "client" more appropriate. We do use the word `application` when talking about that specific section of the Auth0 dashboard below.

File: _posts/2018-06-14-using-m2m-authorization.markdown (hunk 4 above)
Comment (sebadoom) [reply _posts/2018-06-14-using-m2m-authorization.markdown]: If this goes live soon, I'll change this section to describe how to do this with the new M2M UI.

File: _posts/2018-07-05-using-m2m-authorization.markdown [hunk 6]
```diff
@@ -0,0 +1,176 @@
+---
+layout: post
+title: "Using Machine to Machine (M2M) Authorization"
+description: "APIs are first-class citizens at Auth0. Learn how to set up non-interactive apps to perform machine to machine API authorization."
```
Comment (scottmathson-auth0) [inline _posts/2018-07-05-using-m2m-authorization.markdown]: Suggested `description` rewrite to incorporate highly searched, relevant keywords:
 
"How to set up non-interactive apps using the client credentials grant and Auth0 to perform IoT device, CLI tool, and more machine to machine API authorization."

File: _posts/2018-07-05-using-m2m-authorization.markdown [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+{
+  "access_token": "eyJz93a...k4laUWw",
+  "token_type": "Bearer",
+  "expires_in": 86400
+}
+```
+
+In Auth0, to use the client credentials grant, you can create a new "machine to machine" application from the dashboard:
```
Comment (scottmathson-auth0) [inline _posts/2018-07-05-using-m2m-authorization.markdown]: Another great opportunity for internal linking, here.

```In Auth0, to use the client credentials grant, you can [create a new "machine to machine" application from the dashboard](https://auth0.com/docs/applications/machine-to-machine):```

File: _posts/2018-07-05-using-m2m-authorization.markdown [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+![Machine to Machine in the Auth0 Dashboard](https://cdn.auth0.com/blog/m2m/2-create.png)
+
+A machine to machine application requires the selection of at least one API. This is the API that gets selected through the `audience` claim in the HTTP request above.
+
+![Select API](https://cdn.auth0.com/blog/m2m/3-select-api.png)
```
Comment (scottmathson-auth0) [inline _posts/2018-07-05-using-m2m-authorization.markdown]: More context within alt tag here, suggested: `[Select API in the Auth0 Dashboard]`

**Pull request merged**